src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/facet/FacetLoaderService.java

```diff
@@ -0,0 +1,429 @@
+import edu.harvard.dbmi.avillach.dictionaryetl.facetcategory.FacetCategoryModel;
+import edu.harvard.dbmi.avillach.dictionaryetl.facetcategory.FacetCategoryRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.*;
+
+/**
+ * Loads facet categories and facets, persists metadata, and maps concepts.
+ * Now supports OR-grouped expressions while preserving backward compatibility.
+ */
+@Service
+public class FacetLoaderService {
+
+    private final edu.harvard.dbmi.avillach.dictionaryetl.facetcategory.FacetCategoryRepository facetCategoryRepository;
+    private final FacetRepository facetRepository;
+    private final ConceptRepository conceptRepository;
+    private final FacetConceptRepository facetConceptRepository;
+
+    // Meta repositories + object mapper
+    private final FacetMetadataRepository facetMetadataRepository;
+    private final ObjectMapper objectMapper;
+
+    public static final String KEY_FACET_EXPRESSIONS = "facet_loader.expressions";
+    private static final String KEY_FACET_EXPRESSIONS_HASH = "facet_loader.expressions_sha256hex";
+
+    public static final String KEY_EFFECTIVE_EXPRESSIONS = "facet_loader.effective_expressions";
+    private static final String KEY_EFFECTIVE_EXPRESSIONS_HASH = "facet_loader.effective_expressions_sha256hex";
+
+    // New: grouped metadata keys
+    public static final String KEY_FACET_EXPRESSION_GROUPS = "facet_loader.expression_groups";
+    private static final String KEY_FACET_EXPRESSION_GROUPS_HASH = "facet_loader.expression_groups_sha256hex";
+
+    public static final String KEY_EFFECTIVE_EXPRESSION_GROUPS = "facet_loader.effective_expression_groups";
+    private static final String KEY_EFFECTIVE_EXPRESSION_GROUPS_HASH = "facet_loader.effective_expression_groups_sha256hex";
+
+    public FacetLoaderService(FacetCategoryRepository facetCategoryRepository, FacetRepository facetRepository,
+                              ConceptRepository conceptRepository, FacetConceptRepository facetConceptRepository,
+                              FacetMetadataRepository facetMetadataRepository,
+                              ObjectMapper objectMapper) {
+        this.facetCategoryRepository = facetCategoryRepository;
+        this.facetRepository = facetRepository;
+        this.conceptRepository = conceptRepository;
+        this.facetConceptRepository = facetConceptRepository;
+        this.facetMetadataRepository = facetMetadataRepository;
+        this.objectMapper = objectMapper.copy().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+    }
+
+    @Transactional
+    public ClearResult clear(FacetClearRequest request) {
+        long categoriesDeleted = 0;
+        long facetsDeleted = 0;
+        long mappingsDeleted = 0;
+
+        if (request == null) return new ClearResult(0,0,0);
```

Comment:
Bracket all ifs. Also, under what circumstances will `request` be null? Seems like a weird case. I'd log it, and consider whether this code is too defensive. Does a null request represent a valid invocation of this method, or should things fail?